### DISCUSSION TITLE
Body SaveState question

### DISCUSSION THREAD

**Discussion by Ivan-le-sketch** [jrouwe/JoltPhysics#1373] [General]
I'm considering using Jolt for a multiplayer game using some sort of rollback netcode approach and am looking for ways to reduce the amount of data to save/be sent over the wire.

Would it make sense to exclude Body MotionProperties force and torque from the recorded state and ensuring these are always zero before recording ? I'm not very experienced regarding how the engine works but I'm assuming force and torque are zero at the end of the simulation step.

**Comment by jrouwe** [comment]
They are zero after the step indeed, so if you always make a snapshot right after the step then you could leave them out. The rule for `SaveState` is that it should only save properties that are calculated by the engine and not properties that have been supplied by the user through the public API, so they should not have been added in the first place. However, I'm a bit worried that I'll break peoples code in subtle ways if I remove these.

**Comment by Ivan-le-sketch** [comment]
Fair enough. Would you consider adding a way to choose to either include or exclude them ?

**Comment by jrouwe** [comment]
I'd rather not as that makes things messy.

**Comment by TIHan** [comment]
@Ivan-le-sketch Since this is relevant, I recently open sourced my current implementation of the rollback: https://github.com/jrouwe/JoltPhysics/discussions/1057

I manually set the properties from the public API. This may not be the perfect solution for 100% determinism, but it's good enough: https://github.com/TIHan/Evergreen/blob/cf7ae3f0093343465b1dcaef1c6bb4d6f9beab98/src/native/libs/Evergreen.Physics/egJolt.cpp#L1348
